cardano-node/src/Cardano/Tracing/Tracers.hs

```diff
@@ -413,13 +422,18 @@ mkTracers traceConf tracer txsProcessedCounter = do
         meta <- mkLOMeta Critical Confidential
         traceNamedObject tr' (meta, logValue1)
         traceNamedObject tr' (meta, logValue2)
- 
+
         case mempoolEvent of
-          TraceMempoolRemoveTxs txs _ ->
+          TraceMempoolRemoveTxs [] _ -> return ()
+          TraceMempoolRemoveTxs txs _ -> do
             -- TraceMempoolRemoveTxs are previously valid transactions that are no longer valid because of
             -- changes in the ledger state. These transactions are already removed from the mempool,
             -- so we can treat them as completely processed.
-            modifyMVar_ txsProcessedCounter $ \counter -> return $ counter + fromIntegral (length txs)
+            newCounter <- atomicModifyIORef' bcCounters $ \counters ->
+              let newCounter = bcTxsProcessedNum counters + fromIntegral (length txs)
+              in (counters { bcTxsProcessedNum = newCounter }, newCounter)
+            meta' <- mkLOMeta Notice Public
+            traceNamedObject tr' (meta', LogValue "txsProcessedNum" (PureI $ fromIntegral newCounter))
```

Comment:
What is the semantics`bcTxsProcessedNum` - you are counting them only when removed - this needs to be carefully explained so that end users understand this

Reply:
Yes, I explained it in the comments:

-- TraceMempoolRemoveTxs are previously valid transactions that are no longer valid because of
-- changes in the ledger state. These transactions are already removed from the mempool,
-- so we can treat them as completely processed.

When txs were added in the mempool - they are prepared to processing, and only when they are removed from the mempool, it means that they are completely processed.